### ISSUE TITLE
syntax: Parser.Incomplete does not take escaped newlines into account

### ISSUE THREAD

**Issue by rob-myers** [mvdan/sh#307]
Given code represented as a string, I want to know whether it is incomplete or not.
Your new API worked in some cases (first 3 below), but not others (last 3 below).

The incorrect cases work in `gosh`, although the final one `echo foo \`  shows the wrong prompt.

```js
const { syntax } = require("mvdan-sh");

function isIncomplete(code) {
  let parser = syntax.NewParser(), incomplete;
  try {
    parser.Interactive(
      { read: () => code },
      (stmts) => (incomplete = parser.Incomplete()) && false,
    );
  } catch (e) {}
  console.log(`${code} ${incomplete ? 'incomplete' : 'complete'}`);
}

// incomplete (correct)
isIncomplete('echo "foo\n');
// incomplete (correct, string has same value)
isIncomplete('echo \"foo\n');
// complete (correct, quote is escaped)
isIncomplete('echo \\"foo\n');
// complete (incorrect, but gosh correct)
isIncomplete('echo foo |\n');
// complete (incorrect, but gosh correct)
isIncomplete('echo foo &&\n');
// complete (incorrect, gosh shows wrong prompt)
isIncomplete('echo foo \\\n');
```

**Comment by mvdan** [comment]
You're right that some issues remain in `Incomplete` and `Interactive` - thanks for filing these :)

**Comment by mvdan** [comment]
The last case was indeed a bug; see the commit above for the fix. The tests now pass for me, using `mvdan-sh` built on master:

```
$ node test.js
echo "foo
 incomplete
echo "foo
 incomplete
echo \"foo
 complete
echo foo |
 incomplete
echo foo &&
 incomplete
echo foo \
 incomplete
```

The reason that you were seeing the issue with `|` and `&&` is because the fix for #300 was 6ff14a8080445e472ff16e3423c46f4f84f5dfae, which was pushed a day after I did the 0.2.0 JS release. I'll do another release now, after these fixes.

**Comment by mvdan** [comment]
@rob-myers just pushed 0.3.0, which mainly contains the fix for #300 and this fix here.
